standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HiveAlterHandler.java

```diff
@@ -192,12 +197,12 @@ public void alterTable(RawStore msdb, Warehouse wh, String catName, String dbnam
       // 2) the table is not an external table, and
       // 3) the user didn't change the default location (or new location is empty), and
       // 4) the table was not initially created with a specified location
-      if (rename
-          && !oldt.getTableType().equals(TableType.VIRTUAL_VIEW.toString())
-          && (oldt.getSd().getLocation().compareTo(newt.getSd().getLocation()) == 0
-            || StringUtils.isEmpty(newt.getSd().getLocation()))
-          && !MetaStoreUtils.isExternalTable(oldt)) {
-        Database olddb = msdb.getDatabase(catName, dbname);
+      if (replDataLocationChanged
+              || (rename
```

Comment:
This patch is not dealing with set location at source. It deals with changing table type to external at source via ALTER table EXTERNAL=true in table properties. So, this scenario is not valid and not in scope.

Reply:
what i meant is the flow can be kept same for replication and normal flow to avoid adding extra complexity ..

Reply:
It cannot be same because the behaviour is different. If location is set for normal flow, we shouldn't update for partitions. Also, it is risky to modify any of current behaviour. It is not in scope of this ticket.